### PR TITLE
Setup compatibility testing to support the oldest supported libtpu ve…

### PR DESCRIPTION
**Pull request by sagyakwa**
…rsion (12 weeks)

- A new matrix option was added to jaxlib-version matrix
- Expected to immediately fail due to strict jaxlib version check (requires libtpu to have the same API verison as jaxlib, fails otherwise)
- Due to expected failure, so chat will be sent when the jaxlib-version matrix is "nightly+oldest_supported_libtpu"

### CODE REVIEW

File: .github/workflows/cloud-tpu-ci-nightly.yml [hunk 1]
```diff
@@ -24,9 +24,12 @@ jobs:
     strategy:
       fail-fast: false # don't cancel all jobs on failure
       matrix:
-        jaxlib-version: ["pypi_latest", "nightly"]
+        jaxlib-version: ["pypi_latest", "nightly", "nightly+oldest_supported_libtpu"]
         tpu-type: ["v3-8", "v4-8", "v5e-4"]
     name: "TPU test (jaxlib=${{ matrix.jaxlib-version }}, ${{ matrix.tpu-type }})"
+    env:
+      DATE_12_WEEKS_AGO: $(python3 -c "import datetime; print(datetime.date.today() - datetime.timedelta(weeks=12))")
```
Comment (skye) [inline .github/workflows/cloud-tpu-ci-nightly.yml:31]: Let's not actually use the rolling window to start and just specify the exact libtpu-nightly version that should be tested, and we can manually update that to start. We'll probably want to automatically roll the version in the future so we don't have to manually update it, but I wanna start on the conservative side (e.g. to make sure our version checks in the code line up with what we're testing)

File: .github/workflows/cloud-tpu-ci-nightly.yml [hunk 2]
```diff
@@ -56,6 +59,16 @@ jobs:
               -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
             pip install requests
 
+          elif [ "${{ matrix.jaxlib-version }}" == "nightly+oldest_supported_libtpu" ]; then
+          env:
```
Comment (skye) [inline .github/workflows/cloud-tpu-ci-nightly.yml:63]: I don't think you can set an env var this way in the middle of the bash commands to run? Also this needs to be set for the test commands as well. You should just use a conditional in the regular env: block of the install + test steps (https://docs.github.com/en/actions/learn-github-actions/expressions#example)

**Review by skye — CHANGES_REQUESTED** [review]

**Comment by sagyakwa** [comment]
@jying3 cc

File: .github/workflows/cloud-tpu-ci-nightly.yml [hunk 3]
```diff
@@ -24,9 +24,12 @@ jobs:
     strategy:
       fail-fast: false # don't cancel all jobs on failure
       matrix:
-        jaxlib-version: ["pypi_latest", "nightly"]
+        jaxlib-version: ["pypi_latest", "nightly", "nightly+oldest_supported_libtpu"]
         tpu-type: ["v3-8", "v4-8", "v5e-4"]
     name: "TPU test (jaxlib=${{ matrix.jaxlib-version }}, ${{ matrix.tpu-type }})"
+    env:
+      LIBTPU_VERSION_DATE: 20230905
```
Comment (jyingl3) [inline .github/workflows/cloud-tpu-ci-nightly.yml:31]: 20230905 is too old. It needs to be after version 0.29 (https://github.com/openxla/xla/blob/main/xla/pjrt/c/CHANGELOG.md#029-sep-6-2023), and the earliest is 20230907. I think you can also try install and check the PJRT API version of the libtpu (through a jax built from source).

**Review by skye — APPROVED** [review]

**Comment by skye** [comment]
Looks good! We should test this to double check it works, but we can commit and then try :)

**Pull request merged**